x-pack/legacy/plugins/lens/public/datatable_visualization_plugin/visualization.tsx

```diff
@@ -180,7 +181,7 @@ export const datatableVisualization: Visualization<
     const datasource = frame.datasourceLayers[layer.layerId];
     const operations = layer.columns
       .map(columnId => ({ columnId, operation: datasource.getOperationForColumnId(columnId) }))
-      .filter(o => o.operation);
+      .filter((o): o is { columnId: string; operation: Operation } => !!o.operation);
```

Comment:
:information_source: Making this a type guard, makes TS know below that `operation` can't be `null` anymore.